### PR TITLE
Refactor CI infrastructure

### PR DESCRIPTION
**Pull request by bast**
This is work in progress. Feel free to ignore until it converges (will ping you then) but also feel free to watch and comment.

### CODE REVIEW

**Comment by bast** [comment]
@robertodr here we go, ready for review after quite some refactor: https://twitter.com/jeremyjkun/status/927192831902359552

File: .scripts/report_versions.sh [hunk 1]
```diff
@@ -1,16 +1,9 @@
...
-# CMake
-cmake --version
-
-# Generators
-if [[ "${GENERATOR}" == "Unix Makefiles" ]]; then
-    make --version
-else
-    ninja --version
-fi
+for tool in cmake make ninja; do
```
Comment (robertodr) [inline .scripts/report_versions.sh]: :+1:

**Comment by bast** [comment]
Oops - will restore those and resubmit today. Thanks for review!

**Comment by bast** [comment]
I think Travis and Drone are back to normal again. Mac OS X again takes forever.

**Comment by bast** [comment]
There is still a lot to do but at least it's moving forward now. AppVeyor is next and all the other pending things. We can see how we feel about the menu.yml structure. If we don't like it, we will adjust.

**Pull request merged**